pyServer/GuestFishWrapper.py

```diff
@@ -105,6 +107,48 @@ def CreateArchive(self, zipFilename, targetDir):
                         zf.write(path, os.path.relpath(path, base_path))        
         return zipFilename
 
+    def RunCredentialScanner(self, targetDir):
+        if not os.path.exists("/CS_Latest/tools/CredentialScanner.exe"):
+            strMsg = "Credential Scanner executable not found, skipping step."
+            self.rootLogger.warning(strMsg)
+            return
+
+        step_start_time = datetime.now()
+        output_file = targetDir + "/credscan"
+        try:
+            # Run credential scanner
+            bash_command = ["mono", "--runtime=v4.0", "/CS_Latest/tools/CredentialScanner.exe", "-I", targetDir, "-S", "/CS_Latest/tools/Searchers/buildsearchers.xml,/CS_Latest/tools/Searchers/diskinspectsearchers.xml", "-O", output_file, "-v"]
+            result = subprocess.run(bash_command, timeout=600)
+        except subprocess.TimeoutExpired:
+            strMsg = "Credential Scanner timed out after 10 min."
+            self.rootLogger.warning(strMsg)
+        except:
+            strMsg = "Credential Scanner failed to run due to an exception."
+            self.rootLogger.warning(strMsg)
+        else:
+            if result.returncode != 0: 
```

Comment:
do we need an else for this check or is it expected to throw an exception for non-zero?

Reply:
A non-zero return code is not handled as an exception, so it will not be caught by the try-except block and goes into the else. However, this check is needed because it may mean the output files used below have not been created. Does this help clarify?